var/spack/repos/builtin/packages/cudnn/package.py

```diff
@@ -13,39 +13,141 @@ class Cudnn(Package):
 
     homepage = "https://developer.nvidia.com/cudnn"
 
-    version('7.6.3-10.1-x86_64', sha256='352557346d8111e2f954c494be1a90207103d316b8777c33e62b3a7f7b708961',
+    # Latest versions available at:
+    #   https://developer.nvidia.com/rdp/cudnn-download
+    # Archived versions available at:
+    #   https://developer.nvidia.com/rdp/cudnn-archive
+    # Note that download links don't work from command line,
+    # need to use modified URLs like below.
+
+    # 7.6.5
+    version('10.1-linux-x64-v7.6.5.32',
+            sha256='7eaec8039a2c30ab0bc758d303588767693def6bf49b22485a2c00bf2e136cb3',
+            url='https://developer.download.nvidia.com/compute/redist/cudnn/v7.6.5/cudnn-10.1-linux-x64-v7.6.5.32.tgz',
+            preferred=True)
```

Comment:
When we add a `when=` arg to the `version()` directive, we can be more fancy about the preferred version.